app/models.py

```diff
@@ -98,8 +108,33 @@ def delete(self):
         db.session.commit()
 
     @staticmethod
-    def get_all():
-        return Recipes.query.all()
+    def recipe_name_unique(recipe_name, category_id):
+        recipe_existence = Recipes.query.filter_by(recipe_name=recipe_name, category_id=category_id).first()
+        return recipe_existence
+
+    @staticmethod
+    def get_all(category):
+        return Recipes.query.filter_by(category_id=category)
 
     def __repr__(self):
         return "<Recipes: {}>".format(self.recipe_name)
+
+
+class BlacklistToken(db.Model):
```

Comment:
I think we had discussed on you researching on another method of invalidating a token after logout if it still hasn't expired. Were you able to do so?

Reply:
You discussed with Sylvance who approached me of the same but I had tried to figure out another method before which din't work. I approached Hussein and some other people and showed me this method.

Reply:
Ok :+1: